### ISSUE TITLE
Add a "Collapse" in the "folder name" right-click menu

### ISSUE THREAD

**Issue by fredericch** [apache/netbeans#4713]
### Description

This function would collapse recursively the tree structure under the selected folder. It would be implemented in the "Projects" and "Files" window views.

### Use case/motivation

The issue today is that the only function available is "Collapse All".

Sometimes I need to collapse recursively specific folders but not the parts I am currently working on.

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by mbien** [comment]
Not sure if I understand this. You want to close all nodes except those of the files you have currently open?

i usually just use collapse-all and then press ctrl+shift+1/2/3 to open the tree in projects/files/favorites for the active editor tab I am  currently interested in.

**Comment by mbien** [comment]
I see. Like `Collapse All`, but `Collapse` is just for the selected sub-tree, accessible via context menu.

**Comment by WhiteHsu** [comment]
I did development with Netbeans and have interest on this issue. May I take this?  :-)

**Comment by mbien** [comment]
@WhiteHsu of course. Simply open a PR or a draft when ready so that a reviewer can take a look.

**Comment by WhiteHsu** [comment]
Thanks @mbien .

I'm trying to build the project but encounter errors. May I consult your opinions?

What I've done are:
1. Fork and clone the project to my local on my Windows 10.
2. Install JDK 19 (The Latest Version)
3. Download ANT 1.10.12 (Latest Version too)
4. Then go to the local dir of the home folder netbeans
5. Run command "ant" with Admin auth.

Then I got the 
error: package javax.help does not exist
    [javac] import javax.help.HelpSet

May you kindly suggest what I may miss?

Pls refer to my cmd screens as attached. Thank you very much. 

![image](https://user-images.githubusercontent.com/7831418/195999178-8e27cfda-0990-4496-bb7d-62234331ba6b.png)
![image](https://user-images.githubusercontent.com/7831418/195999181-3ed114c2-963e-4c7f-b759-fa727d08ffae.png)
![image](https://user-images.githubusercontent.com/7831418/195999185-1f5456c9-e99c-458a-96e3-d5236dde62e6.png)

**Comment by mbien** [comment]
@WhiteHsu can you move this to a github discussion (or the mailing list)? Its better to talk there instead of using an issue for it.

**Comment by WhiteHsu** [comment]
Thanks @mbien . Have posted it here:
https://github.com/apache/netbeans/discussions/4797

Pls kindly help suggest and feedback. Thank you very much.  :-)

**Comment by cazonda** [comment]
@mbien 
Can you assign me this issue?

**Comment by cazonda** [comment]
I don't understand, @mbien. Are you saying that there is no work to do on this issue or you're saying that I should work on another one?

**Comment by mbien** [comment]
@cazonda sorry, what I mean is that you can work on this issue without asking someone to assign it to you. Its good that you mentioned that you want to work on it though - contributions are indeed welcome :)

**Comment by siddharthkarn2212** [comment]
Hi, I am exploring Apache NetBeans as part of my GSOC preparation.

This issue looks interesting and beginner-friendly.
Before starting, I wanted to ask:
:- Is there any design discussion or preferred approach?
:- Which module/package should I explore first for this feature?

I would appreciate any guidance. Thank you !
